x-pack/plugins/ml/public/application/explorer/add_to_dashboard_control.tsx

```diff
@@ -0,0 +1,339 @@
+          if (swimlaneType === SWIMLANE_TYPE.VIEW_BY) {
+            return {
+              ...config,
+              embeddableConfig: {
+                jobIds,
+                swimlaneType,
+                viewBy,
+                limit,
+              },
+            };
+          }
+          return {
+            ...config,
+            embeddableConfig: {
+              jobIds,
+              swimlaneType,
+            },
+          };
+        });
+
+      try {
+        await dashboardService.attachPanels(item.id, item.attributes, panelsData);
+        setAddedDashboards({ ...addedDashboards, [item.id]: 'success' });
+      } catch (e) {
+        toasts.danger({
+          body: e,
+        });
+      }
+    },
+    [addedDashboards, selectedSwimlanes]
+  );
+
+  const columns: EuiTableProps['columns'] = useMemo(
+    () => [
+      {
+        field: 'title',
+        name: i18n.translate('xpack.ml.explorer.dashboardsTable.titleColumnHeader', {
+          defaultMessage: 'Title',
+        }),
+        sortable: true,
+        truncateText: true,
+      },
+      {
+        field: 'description',
+        name: i18n.translate('xpack.ml.explorer.dashboardsTable.descriptionColumnHeader', {
+          defaultMessage: 'Description',
+        }),
+        truncateText: true,
+      },
+      {
+        field: 'actions',
+        name: i18n.translate('xpack.ml.explorer.dashboardsTable.actionsColumnHeader', {
+          defaultMessage: 'Actions',
+        }),
+        align: 'right',
+        actions: [
+          {
+            name: i18n.translate('xpack.ml.explorer.dashboardsTable.quickAddLabel', {
+              defaultMessage: 'Quick add',
+            }),
+            render: (item) => {
+              return addedDashboards[item.id] === 'pending' ? (
+                <EuiLoadingSpinner size="m" />
+              ) : (
+                <EuiToolTip
+                  content={
+                    addedDashboards[item.id] === 'success' ? (
+                      <FormattedMessage
+                        id="xpack.ml.explorer.dashboardsTitle.successfullyAddedMessage"
+                        defaultMessage="Successfully added"
+                      />
+                    ) : (
+                      <FormattedMessage
+                        id="xpack.ml.explorer.dashboardsTable.quickAddLabel"
+                        defaultMessage="Quick add"
+                      />
+                    )
+                  }
+                >
+                  <EuiButtonIcon
+                    color="primary"
+                    onClick={addSwimlaneToDashboardCallback.bind(null, item)}
+                    iconType={
+                      addedDashboards[item.id] === 'success'
+                        ? 'checkInCircleFilled'
+                        : 'plusInCircle'
+                    }
+                    aria-label="Quick add"
+                    disabled={addedDashboards[item.id] === 'success'}
```

Comment:
Maybe it should just close the modal after adding it to a dashboard, rather than disabling both the add / edit buttons? Otherwise you need to close the modal and re-open if you want to add both an Overall and a View by swimlane to the same dashboard.

Reply:
You can add Overall and View by simultaneously, just make sure both buttons from the group are clicked